urdf_model/include/urdf_model/pose.h

```diff
@@ -168,11 +168,13 @@ class Rotation
 
   void normalize()
   {
-    double s = sqrt(this->x * this->x +
-                    this->y * this->y +
-                    this->z * this->z +
-                    this->w * this->w);
-    if (s == 0.0)
+    const double squared_norm =
+      this->x * this->x +
+      this->y * this->y +
+      this->z * this->z +
+      this->w * this->w;
+
+    if (squared_norm <= 0.0)
```

Comment:
While I understand this is trying to prevent the warning, it feels wrong.  There is no situation in which `squared_norm` could ever be `< 0.0`.

Unfortunately, I have no alternatives to suggest here.  Maybe @scpeters or @sloretz have a suggestion.  If not, then I'll suggest a comment explaining why we are doing this here.

Reply:
> There is no situation in which squared_norm could ever be < 0.0.

I was going to suggest an `assert()`, but it looks like `squared_norm` could be 0.0. That would cause a division by zero when doing `this->x /= s`. With that in mind I think this PR looks reasonable to me.

Reply:
I think adding a comment makes sense.

Reply:
Comment added